pandas/core/common.py

```diff
@@ -472,3 +471,84 @@ def f(x):
         f = mapper
 
     return f
+
+
+def searchsorted_integer(arr, value, side="left", sorter=None):
+    """
+    searchsorted implementation for searching integer arrays.
+
+    We get a speedup if we ensure the dtype of arr and value are the same
+    (if possible) before searchingm as numpy implicitly converts the dtypes
+    if they're different, which would cause a slowdown.
+
+    See :func:`searchsorted` for a more general searchsorted implementation.
+
+    Parameters
+    ----------
+    arr : numpy.array
+        a numpy array of integers
+    value : int or numpy.array
+        an integer or an array of integers that we want to find the
+        location(s) for in `arr`
+    side : str
+        One of {'left', 'right'}
+    sorter : numpy.array, optional
+
+    Returns
+    -------
+    int or numpy.array
+        The locations(s) of `value` in `arr`.
+    """
+    if sorter is not None:
+        sorter = ensure_platform_int(sorter)
+
+    # below we try to give `value` the same dtype as `arr`, while guarding
+    # against integer overflows. If the value of `value` is outside of the
+    # bound of `arr`, `arr` would be recast by numpy, causing a slower search.
+    value_arr = np.array([value]) if is_scalar(value) else np.array(value)
+    iinfo = np.iinfo(arr.dtype)
+    if (value_arr >= iinfo.min).all() and (value_arr <= iinfo.max).all():
+        dtype = arr.dtype
+    else:
+        dtype = value_arr.dtype
+    value = np.asarray(value, dtype=dtype)
+
+    return arr.searchsorted(value, side=side, sorter=sorter)
+
+
+def searchsorted(arr, value, side="left", sorter=None):
+    """
+    Find indices where elements should be inserted to maintain order.
+
+    Find the indices into a sorted array-like `arr` such that, if the
+    corresponding elements in `value` were inserted before the indices,
+    the order of `arr` would be preserved.
+
+    See :class:`IndexOpsMixin.searchsorted` for more details and examples.
+
+    Parameters
+    ----------
+    arr : numpy.array or ExtensionArray
+    value : scalar or numpy.array
+    side : str
+        One of {'left', 'right'}
+    sorter : numpy.array, optional
+
+    Returns
+    -------
+    int or numpy.array
+        The locations(s) of `value` in `arr`.
+    """
+    if sorter is not None:
+        sorter = ensure_platform_int(sorter)
+
+    if is_integer_dtype(arr) and (
+            is_integer(value) or is_integer_dtype(value)):
+        return searchsorted_integer(arr, value, side=side, sorter=sorter)
+    if not (is_object_dtype(arr) or is_numeric_dtype(arr) or
+            is_categorical_dtype(arr)):
+        # E.g. if `arr` is an array with dtype='datetime64[ns]'
+        # and `value` is a pd.Timestamp, we may need to convert value
+        from pandas.core.series import Series
```

Comment:
why are you doing this? this should accept an array, why are you then importing a Series???